app/models/miq_provision_virt_workflow.rb

```diff
@@ -5,46 +5,43 @@ def auto_placement_enabled?
     get_value(@values[:placement_auto])
   end
 
-  def initialize(values, requester, options = {})
-    initial_pass = values.blank?
-    initial_pass = true if options[:initial_pass] == true
-    instance_var_init(values, requester, options)
-
-    # Check if the caller passed the source VM as part of the initial call
-    if initial_pass == true
-      src_vm_id = get_value(@values[:src_vm_id])
-      unless src_vm_id.blank?
-        vm = VmOrTemplate.find_by(:id => src_vm_id)
-        @values[:src_vm_id] = [vm.id, vm.name] unless vm.blank?
-      end
-    end
+  def initial_pass?(values, options)
+    (options[:initial_pass] == true) || values.blank?
+  end
 
-    unless options[:skip_dialog_load] == true
-      # If this is the first time we are called the values hash will be empty
-      # Also skip if we are being called from a web-service
-      @dialogs = get_pre_dialogs if initial_pass && options[:use_pre_dialog] != false
-      if @dialogs.nil?
-        @dialogs = get_dialogs
-      else
-        @running_pre_dialog = true if options[:use_pre_dialog] != false
-      end
-      normalize_numeric_fields unless @dialogs.nil?
+  def load_source_object
+    src_vm_id = get_value(@values[:src_vm_id])
+    unless src_vm_id.blank?
+      vm = VmOrTemplate.find_by(:id => src_vm_id)
+      @values[:src_vm_id] = [vm.id, vm.name] unless vm.blank?
     end
+  end
 
-    password_helper(@values, false) # Decrypt passwords in the hash for the UI
-    @last_vm_id = get_value(@values[:src_vm_id]) unless initial_pass == true
+  def initialize_dialogs(values, options)
+    @dialogs = get_pre_dialogs if options[:skip_dialog_load] != true && initial_pass?(values, options) && options[:use_pre_dialog] != false
 
+    super
+  end
+
+  def configure_dialogs(values, options)
     return if options[:skip_dialog_load] == true
 
-    set_default_values
-    update_field_visibility
+    super
 
     if get_value(values[:service_template_request])
       show_dialog(:requester, :hide, "disabled")
       show_dialog(:purpose,   :hide, "disabled")
     end
   end
 
+  def initialize(values, requester, options = {})
+    super
+
+    load_source_object if initial_pass?(values, options)
```

Comment:
Does this have to happen before the dialogs are loaded? (i.e. in between initialize_dialogs and configure_dialogs?)

Reply:
good question - maybe @lfu knows.

Reply:
See https://github.com/ManageIQ/manageiq/pull/21014#issuecomment-774098443